project/src/Enemies/Pirate/PirateShip.gd

```diff
@@ -9,13 +13,25 @@ export var angular_acceleration_max := 15
 export var angular_drag_factor := 0.1
 export var distance_from_player_min := 200.0
 export var distance_from_obstacles_min := 200.0
+export var aggro_radius := 300.0
+export var distance_from_spawn_max := 600.0
+export var firing_angle_to_player := 4
+export(int, LAYERS_2D_PHYSICS) var projectile_mask := 0
+export var PopEffect: PackedScene
 
 var _acceleration := GSTTargetAcceleration.new()
 var _velocity := Vector2.ZERO
 var _angular_velocity := 0.0
+var _arrive_home_blend: GSTBlend
+var _pursue_face_blend : GSTBlend
+var _current_health := health
 
 onready var agent := GSTSteeringAgent.new()
-onready var player_agent: GSTSteeringAgent = get_tree().get_nodes_in_group("Player")[0].agent
+onready var player_agent: GSTSteeringAgent = (
+		get_tree().get_nodes_in_group("Player")[0].agent
+				if get_tree().get_nodes_in_group("Player").size() > 0
+				else null
```

Comment:
Don't use something like that, it doesn't make sense. First, there's no else clause, so you should just use an if block.
Then, I'm not sure about relying on groups to get a single node ref like the player. Isn't there a better solution?